### PR TITLE
Wireup oracle script data requests

### LINKED ISSUE #1044: Wireup oracle script data requests
Combine all the features we have, including connect with mnemonic, schema data, borsh encoding, sending request transaciton. Make sure if we connect with mnemonic that has BAND, we can send request transaction and everything works properly.

### PR DESCRIPTION
**Pull request by prin-r**
Fix #1044

### CODE REVIEW

File: scan/src/contexts/AccountContext.re [hunk 1]
```diff
@@ -26,11 +35,45 @@ let context = React.createContext(ContextHelper.default);
...
+        | Some({address, privKey}) =>
+          let%Promise data = bandchain->BandWeb3.getAccounts(address |> Address.toBech32);
+          let msgRequest =
+            StdMsgRequest.create(
+              oracleScriptID,
+              ~calldata,
+              ~requestedValidatorCount=4,
+              ~sufficientValidatorCount=4,
+              ~expiration=20,
+              ~prepareGas=20000,
```
Comment (sorawit) [inline scan/src/contexts/AccountContext.re]: @Benzbeeb Example of code where #1115 will break.

File: scan/src/utils/Param.re [hunk 2]
```diff
@@ -0,0 +1,4 @@
+type t = {
```
Comment (sorawit) [inline scan/src/utils/Param.re:1]: If we are having this type. We should change borsh to return this type as opposed to `(string, string)`? @evilpeach @taobun , thoughts?

Comment (sorawit) [inline scan/src/utils/Param.re]: If we are having this type. We should change borsh to return this type as opposed to `(string, string)`? @evilpeach @taobun , thoughts?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/utils/Param.re (hunk 2 above)
Comment (evilpeach) [reply scan/src/utils/Param.re]: Agree to move Param type to `Borsh.re`

**Review by evilpeach — DISMISSED** [review]

File: scan/src/contexts/AccountContext.re [hunk 3]
```diff
@@ -26,11 +35,45 @@ let context = React.createContext(ContextHelper.default);
 let make = (~children) => {
   let (state, dispatch) = React.useReducer(reducer, None);
 
+  let sendRequest =
```
Comment (sorawit) [inline scan/src/contexts/AccountContext.re]: This is bad. Do not send a function to user as a state. Let's have user dispatch action, similar to Connect.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/oracle-script/OracleScriptExecute.re [hunk 4]
```diff
@@ -152,136 +118,44 @@ let resultRender = result => {
...
-  // TODO: wire up later
-  Js.Console.log(code);
-
-  // let (callDataList, setCallDataList) = React.useState(_ => Belt_List.make(numParams, ""));
-
-  // let (result, setResult) = React.useState(_ => Nothing);
+let make = (~id: ID.OracleScript.t, ~schemaOpt: option(string)) => {
+  let (_, dispatch) = React.useContext(AccountContext.context);
+
+  let schema = schemaOpt->Belt_Option.getWithDefault("");
```
Comment (sorawit) [inline scan/src/components/oracle-script/OracleScriptExecute.re]: If there's no schema, we should just show to use that schema is not available. Doing default like this is error prone.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/bindings/BandWeb3.re [hunk 5]
```diff
@@ -44,9 +51,19 @@ let getAccounts = (instance, address) => {
...
+        txHash: rawResponse |> at(["txhash"], string) |> Hash.fromHex,
+        success:
+          switch (
+            rawResponse |> optional(field("logs", list(log => log |> field("success", bool))))
+          ) {
+          | Some(bools) => !bools->Belt_List.some(isSuccess => !isSuccess)
+          | None => false
+          },
+      },
+    ),
```
Comment (evilpeach) [inline scan/src/bindings/BandWeb3.re]: Is it still working on cosmos 0.38 ?

File: scan/src/bindings/StdMsgRequest.re [hunk 6]
```diff
@@ -30,13 +30,13 @@ type t = {
   chain_id: string,
   fee: fee_t,
   memo: string,
-  account_number: string,
+  accountNumber: string,
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re]: Revert this. Don't mess with record that we send to remote endpoint as is.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Comment by sorawit** [comment]
This won't work due until we rechain due to #1225.

**Review by evilpeach — APPROVED** [review]

File: scan/src/pages/OracleScriptIndexPage.re [hunk 7]
```diff
@@ -109,7 +111,20 @@ let make = (~oracleScriptID, ~hashtag: Route.oracle_script_tab_t) =>
...
+               let%Opt paramsInput = schema->Borsh.extractFields("Input");
+               Some(<OracleScriptExecute id=oracleScriptID schema paramsInput />);
+             }
+           ) {
+           | Some(dom) => dom
+           | None =>
+             <div className={Styles.withPadding(20, 20)}>
+               <Text value="Schema not found" color=Colors.gray7 />
+             </div>
+           }
```
Comment (evilpeach) [inline scan/src/pages/OracleScriptIndexPage.re]: move the switch inside `OracleScriptExecute`

File: scan/src/bindings/BandWeb3.re [hunk 8]
```diff
@@ -61,7 +69,17 @@ let broadcast = (instance, signedMsg) => {
   let%Promise rawResponse = instance->_broadcast(signedMsg);
 
   Promise.ret(
-    JsonUtils.Decode.{txHash: rawResponse |> at(["txhash"], string) |> Hash.fromHex},
+    Tx(
+      JsonUtils.Decode.{
+        txHash: rawResponse |> at(["txhash"], string) |> Hash.fromHex,
+        rawLog: rawResponse |> at(["raw_log"], string),
+        success:
+          switch (rawResponse |> optional(field("logs", list(log => log)))) {
```
Comment (evilpeach) [inline scan/src/bindings/BandWeb3.re]: @sorawit Is there a better way to handle this case (for checking log is available)?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/bindings/BandWeb3.re (hunk 8 above)
Comment (sorawit) [reply scan/src/bindings/BandWeb3.re]: Can't we do `rawResponse |> optional(field("logs", _ => ()) |> Belt_Option.isSome`?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/oracle-script/OracleScriptExecuteResponse.re [hunk 9]
```diff
@@ -0,0 +1,142 @@
+module Styles = {
+  open Css;
+
+  let resultWrapper = (w, h, paddingV, overflow_choice) =>
```
Comment (evilpeach) [inline scan/src/components/oracle-script/OracleScriptExecuteResponse.re]: `overflowChioce`

File: scan/src/components/oracle-script/OracleScriptExecute.re [hunk 10]
```diff
@@ -152,197 +120,120 @@ let resultRender = result => {
...
+    {
+      let%Opt schema = schemaOpt;
+      let%Opt paramsInput = schema->Borsh.extractFields("Input");
+      Some(<ExecutionPart id schema paramsInput />);
+    }
+  ) {
+  | Some(dom) => dom
+  | None =>
+    <div className={Styles.withPadding(20, 20)}>
+      <Text value="Schema not found" color=Colors.gray7 />
```
Comment (evilpeach) [inline scan/src/components/oracle-script/OracleScriptExecute.re]: Let's use this pattern.
```
let make = (~id: ID.OracleScript.t, ~schemaOpt) => {
    let%Opt schema = schemaOpt;
    let%Opt paramsInput = schema->Borsh.extractFields("Input");

    <ExecutionPart id schema paramsInput />
} |> Belt.Option.getWithDefault(_, 
  <div className={Styles.withPadding(20, 20)}>
        <Text value="Schema not found" color=Colors.gray7 />
  <div />
)
```

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Review by sorawit — APPROVED** [review]

**Pull request merged**